service/history/shard/context_impl.go

```diff
@@ -2127,6 +2129,16 @@ func newContext(
 			false,
 		)
 	}
+
+	// We need to set lastUpdate time to "now" + "wait between shard updates time" -  "first update interval"
+	// This is done to make sure that first shard update` will happen around "first update interval".
```

Comment:
So lastUpdated will be now + 5m - 10s = now + 4m50s and no updateShard request can go through and write to DB for 9m50s?

I think it should be now - 5m + 10s, then lastUpdate will be now - 4m50s and first update will happen at now + 10s, which is around the first update interval?

Reply:
ops, your right, this is exactly what my intention was, good catch.

"last update time"  = now - "min update interval" + "first update interval".
So
lastUpdates := now -  5m + 10s, and it becomes 4.50 in the past. So first update should happen in 10 sec from now "now"